### PR TITLE
[veigar] Add new port.

### PR DESCRIPTION
**Pull request by winsoft666**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/veigar/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO winsoft666/veigar
+    HEAD_REF master
+    REF 9451b69dd9eb8956f25a1f63f550d80053741bbe
```
Comment (Adela0814) [inline ports/veigar/portfile.cmake:5]:
```suggestion
    REF ${VERSION}
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
@winsoft666  Usage test failed, unable to find header file in the test project.
Repro steps:

1. Install veigar: `./vcpkg install veigar`
2. Create a test cmake project and use the usage provided by vcpkg:
```
     find_package(veigar CONFIG REQUIRED)
     target_link_libraries(mainPRIVATE veigar::veigar)
```
3. Add `#include <veigar/veigar.h>` to cpp file.

**Comment by winsoft666** [comment]
@Adela0814 I think I have fixed this problem, could you take a look?

File: ports/veigar/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DVEIGAR_USE_STATIC_CRT:BOOL=${VEIGAR_USE_STATIC_CRT}
+        -DVEIGAR_BUILD_TESTS:BOOL=OFF
+        -DVEIGAR_BUILD_EXAMPLES:BOOL=OFF
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH share/veigar)
```
Comment (dg0yt) [inline ports/veigar/portfile.cmake:21]:
```suggestion
vcpkg_cmake_config_fixup()
```
Default is `share/${PORT}`, so we don't need to be explicit.

File: versions/v-/veigar.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "versions": [
+    {
+      "git-tree": "055f980031af98f257256c200754062ffe9e6c3c",
+      "version": "1.2",
+      "port-version": 0
+    },
+    {
+      "git-tree": "09ccb0702354b6a33b28656e3b78c36841df8caf",
+      "version": "1.0",
+      "port-version": 0
```
Comment (dg0yt) [inline versions/v-/veigar.json:11]:
```suggestion
```
Only one record can be added in a single PR.

File: ports/veigar/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "veigar",
+  "version": "1.2",
+  "maintainers": "winsoft666 <winsoft666@outlook.com>",
+  "description": "Cross platform RPC library using shared memory.",
+  "homepage": "https://github.com/winsoft666/veigar",
+  "license": "GPL-3.0-or-later",
+  "supports": "(windows | linux) & !uwp",
```
Comment (dg0yt) [inline ports/veigar/vcpkg.json:8]:
```suggestion
  "supports": "linux | (windows & !uwp & !xbox)",
```
if it really doesn't support osx or android.

**Comment by Adela0814** [comment]
Usage test pass on `x64-windows`.

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**